app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,76 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if Helpers.is_logged_in?(session)
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/logout' do
+   session.clear
+   redirect to '/login'
+ end
+
+ patch '/tweets/:id' do
+     @tweet = Tweet.find(params[:id])
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...